packages/skill-lesson/lib/lesson-resource.ts

```diff
@@ -0,0 +1,73 @@
+import groq from 'groq'
+import z from 'zod'
+import {BaseLessonResourceSchema} from '../schemas/base-lesson-resource'
+import {sanityClient} from '../utils/sanity-client'
+
+export const LessonSchema = z
+  .object({
+    _id: z.string().optional(),
+    _key: z.string().optional(),
+    solution: z.nullable(
+      z
+        .object({
+          _key: z.string(),
+        })
+        .merge(BaseLessonResourceSchema)
+        .optional(),
+    ),
+  })
+  .merge(BaseLessonResourceSchema)
```

Comment:
I don't love all the variants of this type of object and the lesson/lesson resource/exercise stuff.

Reply:
technically this is an extremely stripped down generic exercise shape, works for now but definitely something to flag and thing about